lib/snapshot.js

```diff
@@ -0,0 +1,9 @@
+module.exports = Snapshot;
+function Snapshot(collection, id, version, type, data, meta) {
+  if (collection) this.collection = collection;
```

Comment:
I would always assign `collection` and `m` to keep the object shape always the same, which is friendly to the JS optimizer.

Reply:
So I tried that, and then a bunch of tests broke. I can do it, but I personally don't understand the knock-on effects of eg storing `collection` in `MemoryDB`...?

Reply:
Re storing the `collection` field, I agree that it's not necessary and would expect no impact on tests and apps. Maybe for simplicity the tests are just written to check if there are no extra fields beyond those expected (using `expect(...).eql(...)`)?

PS. Here's the first article I found about keeping object shape the same https://ariya.io/2012/02/javascript-object-structure-speed-matters. You can certainly find more detailed explanations.

Reply:
As in you think I _should_ store `collection` in `MemoryDB`?

Reply:
I wouldn't *mind* if it was stored in MemoryDB. It certainly *should not* be stored by *production* DB drivers, as it would be just a waste of space.

If mere presence of the `collection` property is a problem, then I would question whether we should reuse the same class for the UI and MemoryDB.

Reply:
Yeah, I mean I'm happy to revert this change if people don't want it. I'm only doing it because people seemed keen in the PR meeting.